org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/concurrent/IRequestManager.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Avaloq Group AG and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server.concurrent;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.xbase.lib.Functions.Function0;
+import org.eclipse.xtext.xbase.lib.Functions.Function1;
+import org.eclipse.xtext.xbase.lib.Functions.Function2;
+
+/**
+ * @author Rub�n Porras Campo - Initial Contribution and API
+ * 
+ * @since 2.40
+ */
+public interface IRequestManager {
```

Comment:
Now we should start using IRequestManager instead of RequestManager in LanguageServerImpl?

Should this interface be annoted with `@ImplementedBy(AbstractRequestManager.class)`

Reply:
You are right that there should be a binding (to the `RequestManager`, not the `AbstractRequestManager`, the later cannot be injected). I prefer to put the binding in the module and that is part of my updated PR.